fdbclient/include/fdbclient/GetEncryptCipherKeys_impl.actor.h

```diff
@@ -217,6 +217,54 @@ Future<EKPGetBaseCipherKeysByIdsReply> _getUncachedEncryptCipherKeys(Reference<A
 	}
 }
 
+ACTOR template <class T>
+Future<EKPHealthStatus> _getEKPHealthStatusReq(Reference<AsyncVar<T> const> db,
+                                               EncryptKeyProxyHealthStatusRequest request) {
+	Optional<EncryptKeyProxyInterface> proxy = _getEncryptKeyProxyInterface(db);
+	if (!proxy.present()) {
+		// Wait for onEncryptKeyProxyChange.
+		TraceEvent("GetEKPHealthStatusEncryptKeyProxyNotPresent");
+		return Never();
+	}
+	request.reply.reset();
+	try {
+		EKPHealthStatus reply = wait(proxy.get().getHealthStatus.getReply(request));
+		return reply;
+	} catch (Error& e) {
+		TraceEvent("GetEKPHealthStatusCaughtError").error(e);
+		return EKPHealthStatus{ false, false, now() };
+	}
+}
+
+ACTOR template <class T>
+Future<EKPHealthStatus> _getEKPHealthStatusImpl(Reference<AsyncVar<T> const> db) {
+	state EncryptKeyProxyHealthStatusRequest request;
+
+	if (!db.isValid()) {
+		TraceEvent(SevError, "GetEKPHealthStatusServerDBInfoNotAvailable");
+		throw encrypt_ops_error();
+	}
+
+	loop choose {
+		when(EKPHealthStatus reply = wait(_getEKPHealthStatusReq(db, request))) {
+			return reply;
+		}
+		// In case encryptKeyProxy has changed, retry the request.
+		when(wait(_onEncryptKeyProxyChange(db))) {}
+	}
+}
+
+ACTOR template <class T>
+Future<EKPHealthStatus> _getEKPHealthStatus(Reference<AsyncVar<T> const> db) {
+	try {
+		EKPHealthStatus status = wait(timeoutError(_getEKPHealthStatusImpl(db), FLOW_KNOBS->EKP_REQUEST_TIMEOUT));
+		return status;
+	} catch (Error& e) {
+		TraceEvent("EKPHealthStatusError").error(e);
+		return EKPHealthStatus{ false, false, now() };
```

Comment:
why are we absorbing all the errors? we should only absorb `timeout_error` ? For instance: `actor_cancelled` needs to be bubbled up..